server/CQRItems.cpp

```diff
@@ -190,4 +191,96 @@ void CQRItems::info() {
 	log(irId, "info", ttm);
 }
 
+void CQRItems::isRegion() {
+    auto irId = genIntReqId("isRegion");
+
+    sserialize::TimeMeasurer ttm;
+    ttm.begin();
+
+    // set up environment
+    const auto gh = d().completer->store().geoHierarchy();
+    const auto cqs = request().get("q");
+    const auto cqr = d().completer->cqrComplete(cqs, d().treedCQR, d().treedCQRThreads);
+    auto const idxStore = d().completer->indexStore();
+    const auto subGraph = d().completer->ghsg();
+    const auto store = d().completer->store();
+
+    // find all regions in the query
+    std::unordered_set<uint32_t> regions;
+    std::unordered_set<uint32_t> cqrCells;
+    for (sserialize::CellQueryResult::const_iterator it(cqr.begin()), end(cqr.end()); it != end; ++it) {
+        const auto &cellParents = subGraph.cellParents(it.cellId());
+        cqrCells.insert(it.cellId());
+        for (const uint32_t cellParent : cellParents) {
+            if(!regions.contains(cellParent))
+                regions.insert(cellParent);
+        }
+    }
+
+    // find regions with similar item count to the query result
+    typedef std::pair<uint32_t, size_t> Region;
+    auto cqrItemCount = cqr.maxItems();
+    std::vector<Region> potentialRegions;
+
+    for(auto regionId: regions) {
+        size_t regionItemCount = 0;
+        const auto& regionCells = idxStore.at(gh.regionCellIdxPtr(regionId));
+        bool tooManyItems = false;
+        float relativeSize;
+        for(const auto cell: regionCells) {
```

Comment:
Willst Du hier die Überdeckung der einzelnen Region ausrechnen?

Das würde ich tendenziell eher oben machen. Da kannst Du einfach den Item-Count bei den jeweiligen Regionen aufaddieren.

Dann hat man für jede Region eine obere Schranke an den Item-Count. Anschließend kann man dann mit `gh.regionItemsCount(regionId)` den exakten Item-Count der Region holen. Wenn die Schranke dann drüber liegt, dann ist die Region wahrscheinlich vollständig überdeckt. Will man die Region mit vollständiger Überdeckung finden, dann würde ich das nur auf Basis von `CQR::const_iterator::fullMatch` machen. Da kann man dann oben einfach je Region die Anzahl an fm-Zellen holen.

Reply:
Finish Review drücken?

Reply:
Ich merge das mal :) In der Github-PR-Extension fehlt aber irgendwie ein Preview :(